### PR TITLE
Added account endpoint plus some Unit tests

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**

### CODE REVIEW

File: src/main/java/gut/follower/org/Controller/AccountController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    private String checkIfEmailNotTaken(Account account) {
+        return Optional
+                .ofNullable(account.getEmail())
+                .filter(acc -> repository.findByEmail(account.getEmail()) == null)
+                .orElseThrow(() -> new IllegalStateException("Email already taken"));
+    }
+
+    private String getPassword(Account account, Account acc) {
```
Comment (pkprzekwas) [inline src/main/java/gut/follower/org/Controller/AccountController.java]: Make this argument names more verbose. Like new/old account data.

File: src/main/java/gut/follower/org/Controller/AccountController.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public Account getAccount(Principal principal){
+        return Optional
+                .ofNullable(repository.findByUsername(principal.getName()))
+                .orElseThrow(() -> new IllegalStateException("Account does not exist"));
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public Account updateAccount(Principal principal, @RequestBody Account account){
+        return Optional
+                .ofNullable(getAccountByName(principal))
```
Comment (pkprzekwas) [inline src/main/java/gut/follower/org/Controller/AccountController.java]: I would start this Optional by checking new account at first.

```
return Optional
                .ofNullable(account)
                .map(newAcc -> {
                                Account oldAcc = getAccountByName(principal);
                                oldAcc.setPassword(getPassword(account, newAcc));
                                oldAcc.setEmail(getEmail(account, newAcc));
                                return repository.save(acc);
                })
                .orElseThrow(() -> new IllegalStateException("Changes could not be applied"));
```

File: src/main/java/gut/follower/org/Controller/AccountController.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                .map(acc -> {
+                    acc.setPassword(getPassword(account, acc));
+                    acc.setEmail(getEmail(account, acc));
+                    return repository.save(acc);
+                })
+                .orElseThrow(() -> new IllegalStateException("Changes could not be applied"));
+    }
+
+    private String getEmail(Account account, Account acc) {
+        return Optional
```
Comment (pkprzekwas) [inline src/main/java/gut/follower/org/Controller/AccountController.java:42]: If you do as I showed above here you don't need to use Optional cos both values are already not null. Instead use casual if-else statement.

File: src/main/java/gut/follower/org/Controller/AccountController.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                .ofNullable(checkIfEmailNotTaken(account))
+                .orElse(acc.getEmail());
+    }
+
+    private String checkIfEmailNotTaken(Account account) {
+        return Optional
```
Comment (pkprzekwas) [inline src/main/java/gut/follower/org/Controller/AccountController.java:49]: This method can also be replaced with if-else. Cos now you filter with acc without using it which makes it weird :)

File: src/main/java/gut/follower/org/Controller/AccountController.java (hunk 3 above)
Comment (AdamGrzybkowski) [reply src/main/java/gut/follower/org/Controller/AccountController.java:42]: Accounts won't be null but account.getEmail() may retrun null, thats why I used Optional. Do you want me to use 
`if(account.getEmail() != null)...`

?

Comment (pkprzekwas) [reply src/main/java/gut/follower/org/Controller/AccountController.java:42]: But you use it in another method:
`.filter(acc -> repository.findByEmail(account.getEmail()) == null)`
Account won't be null so it wont throw NullPointerException

**Pull request merged**